timestamp-impl/src/main/java/com/palantir/timestamp/PersistentUpperLimit.java

```diff
@@ -23,28 +23,26 @@
 import com.palantir.common.time.SystemClock;
 
 public class PersistentUpperLimit {
+
     private final TimestampBoundStore tbs;
     private final Clock clock;
+    private final TimestampAllocationFailures allocationFailures;
+
     @GuardedBy("this")
     private volatile long cachedValue;
     private volatile long lastIncreasedTime;
 
-    public PersistentUpperLimit(TimestampBoundStore tbs, Clock clock) {
+    public PersistentUpperLimit(TimestampBoundStore tbs, Clock clock, TimestampAllocationFailures allocationFailures) {
         this.tbs = tbs;
         this.clock = clock;
+        this.allocationFailures = allocationFailures;
```

Comment:
Have we had any errors with that? I've worked on projects with and without this pattern and have never seen any difference - if we're shipping code that can have NPEs in wiring then we normally had far more pressing testing issues.

However, this specific field is rarely used, your system could be happy for months before you hit an NPE so it probably makes sense to do the extra validation.